### ISSUE TITLE
Inconsistencies in variable units and BMI implementation

### ISSUE THREAD

**Issue by robertbartel** [NOAA-OWP/cfe#29]
There are inconsistencies in the variable units for some BMI variables.  While there might be others as well, (e.g.,  `DEEP_GW_TO_CHANNEL_FLUX`), the most apparent is with these BMI variables.

| BMI Variable Name | Type | Units |
| ---- | ---- | ---- |
| `RAIN_RATE` | output | `m` |
| `atmosphere_water__liquid_equivalent_precipitation_rate` | input | `kg m-2` |

The listed units are those returned by the CFE implementation of the BMI `Get_var_units` function.

## The problem

1. there are reasons to think these units are inconsistent with the intended units of these variables
2. it turns out that `RAIN_RATE` and `atmosphere_water__liquid_equivalent_precipitation_rate` are literally the same variable in memory, but with different advertised units (see item 1.)

Despite the names, these variables aren't rates based on their units, at least not with respect to time.   This seems likely to be incorrect for `atmosphere_water__liquid_equivalent_precipitation_rate`, and almost certainly is for `RAIN_RATE` (units of just `m` don't make sense for a rate).  Some usage of the variables also is suggestive of a time component. 

If either of these variables do have the correct units advertised, then some additional documentation clarifying the apparent terminology inconsistency would help with confusion.

It's impossible, however, for _both variables_ to have the correct units advertised.  The BMI `Get_value_ptr` function will return `&cfe_ptr->aorc.precip_kg_per_m2` as the pointer for either the `RAIN_RATE` variable or the `atmosphere_water__liquid_equivalent_precipitation_rate` variable.  I.e., both BMI variables are using the same memory address, so they are in fact the same thing.  This behavior by itself is valid, but it doesn't make sense with the current unit metadata.

**Comment by jmframe** [comment]
The Get_value_ptr for RAIN_RATE is not implemented correctly, it should return cfe_ptr->timestep_rainfall_input_m, rather than cfe_ptr->aorc.precip_kg_per_m2. Although I am not sure why CFE should have RAIN_RATE in the output variables...

**Comment by SnowHydrology** [comment]
Is there a need for RAIN_RATE to be listed as a BMI output var? If not, it can be removed. Additionally, please note the change in the BMI specification so that precipitation is now given as mm h<sup>-1</sup>: https://github.com/NOAA-OWP/cfe/blob/e8c2d4b0391cec7d4191beb16f8c10650e3e02e2/src/bmi_cfe.c#L222

**Comment by stcui007** [comment]
Yes. It is helpful for comparing how the various calculated output flows
respond to the original input.

**Comment by SnowHydrology** [comment]
@stcui007 that makes sense. Is there a reason you can't use `atmosphere_water__liquid_equivalent_precipitation_rate` or does the model engine only expose the output variables for writing simulation data?

**Comment by stcui007** [comment]
@Keith Jennings - NOAA Affiliate ***@***.***> In the config
file, there is this line:
 "atmosphere_water__liquid_equivalent_precipitation_rate":    "precip_rate",
I assume RAIN_RATE is equivalent to precip_rate? Last time when RAIN_RATE
was removed, it caused a runtime error in running ngen and Jonathan had to
put it back. Perhaps we don't want to spend too much time on it for now?
Anyway, we need RAIN_RATE or its equivalent in the same output file for
comparison and easy post processing.
